File: src/gaetmin.java
import  java.util.concurrent.TimeUnit;
import java.util.*;

public class gaetmin {
	
	public static boolean graine() {
		boolean bit;
		long bite=System.nanoTime() % 2;
		if(bite==1) bit=true;
		else bit=false;
		return bit;
	}
	
	public static boolean rand() {
		boolean bit;
		//long bite = Math.round(Math.random());
	    Random rand = new Random();
	    long bite =rand.nextInt(2);
		if(bite==1)bit=true;
		else bit=false;
		return bit;
	}
	
	public static double esperance(boolean t[]) {
		int j=0;
		 for(int i=0;i<t.length;i++){
			 if(t[i]){j++;}
		 }
		double espe=(double)j/t.length;
		return espe;
	}
	
	public static double variance(boolean t[]) {
		int j=0;
		 for(int i=0;i<t.length;i++){
			 if(t[i]){j++;}
		 }
		double vari=j*(1-esperance(t))*(1-esperance(t))/t.length;
		return vari;
	}
	
	 public static void main(String[] args)
		{System.out.println("Début du programme");
		System.out.println();
		int nb=1000;
		System.out.println("Génération de "+nb+" bits aléatoires");
		System.out.println();

		System.out.println("Méthode gaetmin : ");
		System.out.println();

		long debut=System.nanoTime();
		boolean[] tab=new boolean[nb];
		 for(int i=0;i<nb;i++){
			 tab[i]=graine();
		 }
		 long duree=System.nanoTime()-debut;
		 duree=TimeUnit.MICROSECONDS.convert(duree, TimeUnit.NANOSECONDS);
		 for(int i=0;i<nb;i++){
			 System.out.print(tab[i]+" ");

		 }
		 System.out.println();
		 System.out.println();
		 System.out.println("Temps de génération :"+duree+" microsecondes");
		 System.out.println("Espérance(True) :"+esperance(tab));
		 System.out.println("Variance(True) :"+variance(tab));
		 System.out.println();
		 System.out.println("Méthode classique : ");
		 System.out.println();

		 debut=System.nanoTime();
		 for(int i=0;i<nb;i++){
				 tab[i]=rand();
			 }
		 duree=System.nanoTime()-debut;
		 duree=TimeUnit.MICROSECONDS.convert(System.nanoTime()-debut, TimeUnit.NANOSECONDS);
		 for(int i=0;i<nb;i++){
				 System.out.print(tab[i]+" ");
			 }
		 System.out.println();
		 System.out.println();
		 System.out.println("Temps de génération :"+duree+" microsecondes");
		 System.out.println("Espérance(True) :"+esperance(tab));
		 System.out.println("Variance(True) :"+variance(tab));
		 System.out.println();
			 
		 System.out.println("Fin du programme");
		}
}

File: src/gaetmin.cpp
//============================================================================
// Name        : gaetmin.cpp
// Author      : yves Mercadier
// Version     :
// Copyright   : Your copyright notice
// Description : générateur aléatoire à partir d'une horloge Ansi-style
//============================================================================
#include <iostream>
#include <string>
#include <fstream>
#include <sys/time.h>
#include <math.h>

using namespace std;
int nb=1000;

int graine() {
	struct timeval temps;
    gettimeofday(&temps,NULL);
	int bit=(int(double(temps.tv_usec)*M_PI))%2;
	return bit;
	}

double esperance(int tab[]){
	int j=0;
	for(int i=0;i<nb;i++){
			 if(tab[i]==1){j++;}
		 }
	double espe=(double)j/nb;
	return espe;
	}
	
double variance(int tab[]){
	int j=0;
	for(int i=0;i<nb;i++){
			 if(tab[i]==1){j++;}
		 }
	double vari=j*(1-esperance(tab))*(1-esperance(tab))/nb;
	return vari;
	}
	
int main()
{
	cout << "Début du programme."<< endl;
	cout << "Génération de "<<nb<<" bits aléatoires."<< endl;

	struct timeval debut,fin;
    int tab [nb];
    gettimeofday(&debut,NULL);
	for(int i=0;i<nb;i++){
		tab[i]=graine();
	}
	gettimeofday(&fin,NULL);
	double duree=fin.tv_usec-debut.tv_usec;
	cout << "Temps de génération :"<<duree<<" microsecondes"<< endl;
	cout << "Espérance(True) :"<<esperance(tab)<< endl;
	cout << "Variance(True) :"<<variance(tab)<< endl;

	for(int i=0;i<nb;i++){
		cout << tab[i]<< " ";
	}
    cout << endl;
    cout << "Fin du programme."<< endl;
}
